components/atom/button/src/index.js

```diff
@@ -45,19 +56,38 @@ const AtomButton = (props) => {
     className
   )
   const newProps = cleanProps(props)
-  return (<button {...newProps} className={classNames} disabled={disabled}>
-    {leftIcon && <span className={`${CLASS}-leftIcon`}>{leftIcon}</span>}
-    {leftIcon || rightIcon
-      ? <span className={`${CLASS}-text`}>{children}</span>
-      : children
-    }
-    {rightIcon && <span className={`${CLASS}-rightIcon`}>{rightIcon}</span>}
-  </button>)
+
+  const Button = ({ children, url, disabled, ...attrs }) => link
+    ? <Link {...attrs} href={url}>{children}</Link>
```

Comment:
What do we add the "link" type AND the linkFactory on top of it.
I would either provide a elementFactory to what we already had OR implement the link prop.
Do you see cases where linkFactory would be really needed if we already inherited all the props... ?